### PR TITLE
CreatePolicyRoom Client-Side API

### PR DESCRIPTION
**Pull request by jasperhuangg**
### Details
Add an API function and Report Action for the new CreatePolicyRoom API command.

### Fixed Issues
$ https://github.com/Expensify/Expensify/issues/176849

### Tests
1. Call the API function using the following snippet, then verify that we successfully create the policy room and get a reportID back.
  ```
  API.CreatePolicyRoom({policyID: '<some policyID>', reportName: '#reportName', visibility: 'private'})
      .then(console.log);
  ```

### Tested On

- [ ] Web
- [ ] Mobile Web
- [ ] Desktop
- [ ] iOS
- [ ] Android

### CODE REVIEW

File: src/libs/actions/Report.js [hunk 1]
```diff
@@ -1398,6 +1398,33 @@ function handleInaccessibleReport() {
...
+function createPolicyRoom(policyID, reportName, visibility) {
+    return API.CreatePolicyRoom({policyID, reportName, visibility})
+        .then((response) => {
+            if (!response) {
+                return;
+            }
+            if (response.jsonCode !== 200) {
+                console.error(response.message);
+                return;
+            }
```
Comment (thienlnam) [inline src/libs/actions/Report.js:1417]: Is a missing response ever going to occur? I think we can just make this and remove the !response

if (response && response.jsonCode !== 200)

File: src/libs/actions/Report.js [hunk 2]
```diff
@@ -1398,6 +1398,33 @@ function handleInaccessibleReport() {
...
+            return fetchChatReportsByIDs([response.reportID]);
+        })
+        .then(([{reportID}]) => {
+            if (!reportID) {
+                console.error('Unable to grab policy room after creation');
```
Comment (thienlnam) [inline src/libs/actions/Report.js:1422]: This needs a return

**Review by thienlnam — CHANGES_REQUESTED** [review]

File: src/libs/API.js [hunk 3]
```diff
@@ -358,6 +358,20 @@ function CreateChatReport(parameters) {
...
+ * @param {Object} parameters
+ * @param {String} parameters.policyID
+ * @param {String} parameters.reportName
+ * @param {String} parameters.visibility
+ * @return {Promise}
+ */
+function CreatePolicyRoom(parameters) {
+    const commandName = 'CreatePolicyRoom';
+    requireParameters(['policyID', 'reportName', 'visibility'],
+        parameters, commandName);
```
Comment (Julesssss) [inline src/libs/API.js:371]: NAB: Any reason for the newline here?

Comment (jasperhuangg) [reply src/libs/API.js:371]: nopee, removed!

File: src/libs/actions/Report.js [hunk 4]
```diff
@@ -269,7 +269,7 @@ function fetchIOUReport(iouReportID, chatReportID) {
         }
         return getSimplifiedIOUReport(iouReportData, chatReportID);
     }).catch((error) => {
-        Log.hmmm('[Report] Failed to populate IOU Collection:', error.message);
+        console.debug(`[Report] Failed to populate IOU Collection: ${error.message}`);
```
Comment (thienlnam) [inline src/libs/actions/Report.js:272]: Forgot to update these back 😅

File: src/libs/actions/Report.js [hunk 5]
```diff
@@ -380,16 +380,14 @@ function fetchChatReportsByIDs(chatList, shouldRedirectIfInaccessible = false) {
...
         .catch((err) => {
-            if (err.message !== CONST.REPORT.ERROR.INACCESSIBLE_REPORT) {
-                return;
+            if (err.message === CONST.REPORT.ERROR.INACCESSIBLE_REPORT) {
+                // eslint-disable-next-line no-use-before-define
+                handleInaccessibleReport();
             }
-
-            // eslint-disable-next-line no-use-before-define
-            handleInaccessibleReport();
```
Comment (thienlnam) [inline src/libs/actions/Report.js:392]: Are there any other expected failures than err.message === CONST.REPORT.ERROR.INACCESSIBLE_REPORT?

File: src/libs/actions/Report.js [hunk 6]
```diff
@@ -909,22 +907,16 @@ function fetchActions(reportID, offset) {
             removeOptimisticActions(reportID);
 
             const indexedData = _.indexBy(data.history, 'sequenceNumber');
+            const maxSequenceNumber = _.chain(data.history)
+                .pluck('sequenceNumber')
+                .max()
+                .value();
+
             Onyx.merge(`${ONYXKEYS.COLLECTION.REPORT_ACTIONS}${reportID}`, indexedData);
+            Onyx.merge(`${ONYXKEYS.COLLECTION.REPORT}${reportID}`, {maxSequenceNumber});
```
Comment (thienlnam) [inline src/libs/actions/Report.js:916]: What are these changes for?

File: src/libs/actions/Report.js [hunk 7]
```diff
@@ -1149,17 +1143,15 @@ function deleteReportComment(reportID, reportAction) {
...
+            if (response.jsonCode !== 200) {
+                // Reverse Optimistic Response
+                reportActionsToMerge[sequenceNumber] = {
+                    ...reportAction,
+                    message: oldMessage,
+                };
 
-            Onyx.merge(`${ONYXKEYS.COLLECTION.REPORT_ACTIONS}${reportID}`, reportActionsToMerge);
+                Onyx.merge(`${ONYXKEYS.COLLECTION.REPORT_ACTIONS}${reportID}`, reportActionsToMerge);
+            }
```
Comment (thienlnam) [inline src/libs/actions/Report.js:1154]: Same here, curious what these changes are for

**Review by thienlnam — CHANGES_REQUESTED** [review]

**Pull request closed without merging**